File: benchmarks/NSubstitute.Analyzers.Benchmarks.CSharp.Source/Models/IFoo.cs
using NSubstitute.Core;

namespace NSubstitute.Analyzers.Benchmarks.CSharp.Source.Models
{
    public interface IFoo
    {
        void VoidReturningMethod();

        IFoo ObjectReturningMethod();

        ConfiguredCall CallInfoReturningMethod();

        ConfiguredCall ConfiguredCallReturningProperty { get; }

        ConfiguredCall this[int a, int b, int c, int d] { get; }

        void VoidReturningMethodWithArguments(int x, int y, decimal z);

        IFoo ObjectReturningMethodWithArguments(int x, int y, decimal z);

        IFoo ObjectReturningMethodWithRefArguments(ref int x, ref int y, ref decimal z);

        int Property { get; }

        IFoo ObjectReturningProperty { get; }

        int this[int i] { get; }

        IFoo this[int i, int y] { get; }
    }
}

File: benchmarks/NSubstitute.Analyzers.Benchmarks.VisualBasic.Source/Models/IFoo.vb
Imports NSubstitute.Core

Namespace Models
    Public Interface IFoo
        Sub VoidReturningMethod()
        Function ObjectReturningMethod() As IFoo
        Function CallInfoReturningMethod() As ConfiguredCall
        ReadOnly Property ConfiguredCallReturningProperty As ConfiguredCall
        Default ReadOnly Property Item(ByVal a As Integer, ByVal b As Integer, ByVal c As Integer, ByVal d As Integer) As ConfiguredCall
        Sub VoidReturningMethodWithArguments(ByVal x As Integer, ByVal y As Integer, ByVal z As Decimal)
        Function ObjectReturningMethodWithArguments(ByVal x As Integer, ByVal y As Integer, ByVal z As Decimal) As IFoo
        Function ObjectReturningMethodWithRefArguments(ByRef x As Integer, ByRef y As Integer, ByRef z As Decimal) As IFoo
        ReadOnly Property [Property] As Integer
        ReadOnly Property ObjectReturningProperty As IFoo
        Default ReadOnly Property Item(ByVal i As Integer) As Integer
        Default ReadOnly Property Item(ByVal i As Integer, ByVal y As Integer) As IFoo
    End Interface
End Namespace
